src/jit/emitxarch.cpp

```diff
@@ -214,7 +217,8 @@ bool TakesRexWPrefix(instruction ins, emitAttr attr)
     // cased here.
     //
     // Rex_jmp = jmp with rex prefix always requires rex.w prefix.
-    if (ins == INS_movsx || ins == INS_rex_jmp)
+    // vpermq requires w bit to set to 1
+    if (ins == INS_movsx || ins == INS_rex_jmp || ins == INS_vpermq)
     {
```

Comment:
Thanks for pointing it out. Yes, vpermq needs VEX w1 bit. How about creating two functions named TakesVexWPrefix() and AddVexWPrefix() to handle VEX?